### ISSUE TITLE
Charging Vehicle Battery as a settlement backlog task or a manual task ?

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#889]
**Is your feature request related to a problem? Please describe.**
- Should we treat charging a vehicle battery prior to excursion as a settlement backlog task or a task to be done automatically ?
- Recently, we added the function of battery in a vehicle in #795. However, it would only be recharged via regenerative braking.

**Describe the solution you desire to see**
- If the vehicle is on maintenance mode or is being reserved for a mission, the settlement should bring the vehicle in and recharge its battery using the task backlog.

**Describe the rationale**
- A vehicle's battery has the advantage of providing short-distance travel without having to bother with burning methanol. 

**Specifications  (please complete the following information):**
- pre 3.6.0 build 8074

**Additional context**
- Should the energy stored in the battery be converted to the *equivalent*  # of miles a vehicle can travel, when calculating how much methanol should be pumped onboard ?
- In this case, the vehicle loading class should be adjusted in response to the amount of energy stored in the battery.

**Comment by mokun** [comment]
@bevans2000 

Can this be turned into a settlement backlog task ?

**Comment by bevans2000** [comment]
Certainly we can do it.

**Comment by mokun** [comment]
By the way, it is still a tricky business to estimate mileage when planing for a mission trip. 

One reason that the estimated fuel economy (FE) or fuel consumption (FE) are a bit off sometimes is that in the past, they used to account for only how much methane or methanol are being used per km. 

But after we have added the contribution of energy stored in batteries, the FE and FC are even are off.

In other words, the rovers in mars-sim are actually "hybrid" vehicles and the calculation of FC and FEncan be a challenge since you can define them in various ways. 

That's why currently we implemented 5 types of FC and 5 types of FE for comparisons.

**Comment by mokun** [comment]
@bevans2000,
 
Btw, Charging a vehicle should look like a vehicle maintenance task but need to be separate.
